### PR TITLE
DAOS-6540 dfuse: Do not allow setting of internal xattrs and add test.

### PR DESCRIPTION
**Pull request by ashleypittman**
Multi-user support and other features in dfuse will use the user.dfuse
prefix to extended attribures, so disallow POSIX clients from setting

### CODE REVIEW

**Review by jolivier23 — DISMISSED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -151,7 +151,7 @@ String unit_packages() {
                            'libpmemobj protobuf-c ' +
                            'spdk-devel libfabric-devel ' +
                            'pmix numactl-devel ' +
-                           'libipmctl-devel ' +
+                           'libipmctl-devel python36-pyxattr ' +
```
Comment (mchaarawi) [inline Jenkinsfile:154]: if we are adding a dependency we should update the docker files:
https://github.com/daos-stack/daos/tree/master/utils/docker

**Review by mchaarawi — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (ashleypittman) [reply Jenkinsfile:154]: I added it to the CentOS one already as that's used for pylint and in CI but I haven't added it to the others.  I doubt that ubuntu and centos 8 have enough packages to run testing in master anyway but I'll check.

Comment (mchaarawi) [reply Jenkinsfile:154]: ah i missed the centos7 one. but anyway it's not about CI testing, but many users refer to those for installing dependencies, so we should keep them all up to date.

Comment (ashleypittman) [reply Jenkinsfile:154]: Yes, this will only affect the ability to run NLT.  I'm running a test now but it'll take a while to build all the images.

Comment (ashleypittman) [reply Jenkinsfile:154]: Done.  I had to add PyYAML on a couple of them as well.

**Review by mchaarawi — DISMISSED** [review]

**Review by jolivier23 — APPROVED** [review]

**Review by bfaccini — APPROVED** [review]

**Comment by ashleypittman** [comment]
This failed in DaosVol and IOR with hdf5 tests, is it OK to force-land or should I merge with master?

**Comment by mchaarawi** [comment]
> This failed in DaosVol and IOR with hdf5 tests, is it OK to force-land or should I merge with master?

those failures i assume are bec of a change force landed by Liang to change the oclass table but there was a period of inconsistency as the daos-vol was not built with the DAOS master that had those changes.
IMO it is safe to land this PR.
(technically though the vol tests use dfuse / xattr from the DUNS calls)

**Review by jolivier23 — APPROVED** [review]
Since it's awaiting testing to complete, I'm just removing gatekeeper with this comment

**Comment by ashleypittman** [comment]
I don't know what the exact issue is but dfuse shutdown has been getting slower and this PR has triggered a failure (dfuse has open files when it receives the shutdown request) so I've backported a change to use the background option and sleep between retries.  I've not been able to reproduce locally but the fix makes sense and works in CI.

**Review by mchaarawi — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**